projects/components/src/viewer/code-viewer/code-viewer.component.ts

```diff
@@ -145,14 +159,69 @@ export class CodeViewerComponent implements AfterViewInit, OnChanges, OnDestroy
     this.domMutationObserver.disconnect();
   }
 
+  public get isCodeEmpty(): boolean {
+    return isEmpty(this.code);
+  }
+
   public isLineHighlighted(lineNum: number): boolean {
     return (
       !isEmpty(this.highlightText) && this.codeLines[lineNum].toLowerCase().includes(this.highlightText.toLowerCase())
     );
   }
 
-  public onSearch(searchText: string): void {
-    this.searchText = searchText;
+  public onSearch(searchString: string): void {
+    // For case-insensitive search
+    const searchText = searchString.toLowerCase();
+    const codeTexts = this.codeTexts.map(text => text.toLowerCase());
+
+    // Remove existing child background elements
+    this.codeLineBackgroundElements.forEach(
+      codeLineBackgroundElement => (codeLineBackgroundElement.nativeElement.innerHTML = '')
+    );
+
+    const searchLen: number = searchText.length;
+    if (searchLen === 0) {
+      return;
+    }
+
+    const searchedPositions: Position[][] = Array.from(Array(codeTexts.length), () => []);
+
+    // Get all searched text positions
+    codeTexts.forEach((codeText, index) => {
+      for (let i = 0; i <= codeText.length - searchLen; i++) {
+        const searchedIndex = codeText.indexOf(searchText, i);
+        if (searchedIndex > -1) {
+          searchedPositions[index].push({ start: searchedIndex, end: searchedIndex + searchLen });
+          i = searchedIndex + searchLen - 1;
+        } else {
+          break;
+        }
+      }
+    });
+
+    // Add background elements for searched positions
+    this.codeLineBackgroundElements.forEach((codeLineBackgroundElement, index) => {
+      codeLineBackgroundElement.nativeElement.appendChild(this.getBackgroundElements(searchedPositions[index]));
+    });
+  }
+
+  // Background elements for search
+  private getBackgroundElements(positions: Position[]): DocumentFragment {
+    const backgroundElemDocFragment: DocumentFragment = document.createDocumentFragment();
+
+    positions.forEach(position => {
+      const backgroundElem = document.createElement('div');
+      backgroundElem.className = 'bg-searched';
+      backgroundElem.style.height = '100%';
```

Comment:
you can possibly put height, background color and position in `bg-searched` class